parser/clj_kondo/impl/rewrite_clj/node/protocols.clj

```diff
@@ -43,7 +43,7 @@
 (defn ^:no-doc concat-strings
   "Convert nodes to strings and concatenate them."
   [nodes]
-  (reduce str (map string nodes)))
+  (string/join " " (map string nodes)))
```

Comment:
Although the tests passed, I was a little hesitant to change this function because of potential breakage. If that's an issue, I can just make this a separate function and call it only for `SeqNode`s.

Reply:
Yeah, please don't change it here. Made a custom print function for the lint output.

Reply:
Although I don't think it should break anything actually. It might make sense to leave it here... I'll have to think about it. This might actually be an improvement ;).

Reply:
@jysandy To be sure, let's not change the vendored version of rewrite-clj but make a custom printer that does what you did there.

Reply:
I took a cursory look at the callers of this function, and I think it might actually be an improvement. That being said, I've made a new printer instead in the latest commit like you asked.

Reply:
@jysandy What would be the consequences if we changed the rewrite-clj function? What are the callers and where does this improve things? I have something in mind, but I wonder if you have the same in mind.

Reply:
These are the callers according to Cursive:

![Screenshot 2021-01-12 at 12 39 30 PM](https://user-images.githubusercontent.com/8731801/104281244-3a6d3400-54d3-11eb-887c-f9a6010d6f9f.png)

For `fn`, `forms`, `meta` and `seq`, I think it makes sense to have the children separated by spaces when printed, so that they are visually distinct. For `quote`, `reader_macro` and `uneval`, I don't think these nodes should have more than one child anyway. But I could be wrong.

Reply:
I don't think clj-kondo relies a specific form of the printed form but it may help debugging in hooks code when you call `println` or `str` on generated hook code. So all in all, I think this will be an improvement. Let's try it and see if the tests pass. You can temporarily revert your custom print function.

Reply:
@borkdude I've removed the custom printer in the latest commit.